conformance/lang/expressions/logical-expr/logical_or_expr.balt

```diff
@@ -0,0 +1,853 @@
+    var b = a || true;
+    boolean _ = b;
+    io:println(b is boolean); // @output true
+    var c = getExpr() || b;
+    boolean _ = c;
+    io:println(c is boolean); // @output true
+    var d = false || 4 != 5;
+    boolean _ = d;
+    io:println(d is boolean); // @output true
+}
+
+function getExpr() returns boolean {
+    return true;
+}
+
+Test-Case: error
+Description: Test static type of the result of a logical-expr by assigning it to an invalid type.
+Labels: logical-expr, is-expr, var, equality, function-call-expr
+
+function errorFunction() {
+    var a = true || false;
+    int _ = a; // @error static type of logical-expr is boolean which is incompatible with an integer
+    var b = a || true;
+    int _ = b; // @error static type of logical-expr is boolean which is incompatible with an integer
+    var c = getExpr() || b;
+    int _ = c; // @error static type of logical-expr is boolean which is incompatible with an integer
+    var d = false || 4 != 5;
+    int _ = d; // @error static type of logical-expr is boolean which is incompatible with an integer
+}
+
+function getExpr() returns boolean {
+    return true;
+}
+
+Test-Case: output
+Description: Test static type of expressions in logical expression using subtypes of boolean expression.
+Labels: logical-expr, is-expr
+
+function init() {
+    boolean & readonly a = true;
+    boolean d = a || true;
+    io:println(a is boolean); // @output true
+}
+
+Test-Case: error
+Description: Test static type of expression in logical expression via invalid types using subtypes of boolean expression.
+Fail-Issue: ballerina-platform/ballerina-lang#35693
+Labels: logical-expr, module-type-defn, union-type
+
+type boolean_false false;
+
+type boolean_true true;
+
+type boolean_sub true|int;
+
+function errorFunction() {
+    boolean_true a = true;
+    boolean_false b = false;
+    boolean_sub z = true;
+    boolean c = a || b; // @error since static type of each expression is boolean this is a valid logical expression
+    boolean d = b || false; // @error since static type of each expression is boolean this is a valid logical expression
+    boolean e = z || true; // @error since static type of each expression is boolean this is a valid logical expression
+    boolean f = a || false; // @error since static type of each expression is boolean this is a valid logical expression
+}
+
+Test-Case: output
+Description: Test if static type of result of a logical-expr is boolean in relation to expression evaluation in rhs and lhs expression.
+Labels: logical-expr, is-expr
+
+function init() {
+    boolean a = true || true;
+    io:println(a is true); // @output true
+
+    a = true || false;
+    io:println(a is true); // @output true
+}
+
+Test-Case: output
+Description: Test logical expression in module level.
+Labels: logical-expr, is-expr
+
+boolean a = true || false;
+boolean b = true || true;
+boolean c = false || false;
+boolean d = false || true;
```

Comment:
Why module-level variables?

Reply:
Because it is testing logical expression in module level.